### ISSUE TITLE
[jest-dom/prefer-in-document] False positive for property of HTMLElement

### ISSUE THREAD

**Issue by kenlyon** [testing-library/eslint-plugin-jest-dom#414]
**Repro:**

I want to confirm that a particular button has the `title` attribute set.

Here's the simplest way to reproduce it:

```
render(<button title="hello there" />);
const firstButton = screen.getAllByRole("button")[0];
const iconId = firstButton.title;
expect(iconId).toBeDefined();
```

**Expected:**

This should be valid. I'm not trying to assert that the button is in the DOM. I know it came from the DOM, but I want to confirm one of its attributes is set.

**Actual:**

The `toBeDefined` method has the error: 

_Prefer .toBeInTheDocument() for asserting DOM node existence_

**Workaround:**

If you create a new string that's not a property of the `HTMLElement`, the error goes away:

```diff
-expect(iconId).toBeDefined();
+expect(iconId + "").toBeDefined();
```

**Comment by kenlyon** [comment]
Thanks for the suggestion. The `toHaveAttribute` method works in my situation.
